### PR TITLE
lego: 3.0.2 -> 3.1.0

### PR DESCRIPTION
**Pull request by wizeman**
###### Motivation for this change

lego 3.0.2 won't build anymore because one of the go dependencies cannot be downloaded due to the commit being missing.

This should probably be backported to 19.09.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @andrew-d

### CODE REVIEW

File: pkgs/tools/admin/lego/default.nix [hunk 1]
```diff
@@ -2,16 +2,16 @@
...
   src = fetchFromGitHub {
     owner = "go-acme";
     repo = pname;
     rev = "v${version}";
-    sha256 = "0q73522yblcjsyscsppwnxfw6m249zr9whb93bhv5i5z012gy6mx";
+    sha256 = "00m2w987czqj3l94834fpzc019pqi60jrqk4cnlfnc5mxjb0dbq1";
   };
 
-  modSha256 = "00pl8l8h01rfxyd0l4487x55kfqhpm0ls84kxmgz3vph7irm6hcq";
+  modSha256 = "0jhhzwz0z8wg9a6y0r32vsigzxblzwqykjgkmwrik06kph0dz7dc";
```
Comment (andrew-d) [inline pkgs/tools/admin/lego/default.nix]:
```suggestion
  modSha256 = "0k3p11cji3p4nzr8aia8hp01wyx1qfx84259dwbfwg1b32ln8rkc";
```

**Review by andrew-d — CHANGES_REQUESTED** [review]
Looks like you've got a hash mismatch:

```
hash mismatch in fixed-output derivation '/nix/store/dmx51z9kdsd4cp3jc47ml7rm9sp8hz64-lego-3.1.0-go-modules':
  wanted: sha256:0jhhzwz0z8wg9a6y0r32vsigzxblzwqykjgkmwrik06kph0dz7dc
  got:    sha256:0k3p11cji3p4nzr8aia8hp01wyx1qfx84259dwbfwg1b32ln8rkc
```

**Comment by aanderse** [comment]
ping @wizeman

**Comment by wizeman** [comment]
I fixed the hash but note that the `modSha256` I had previously was the one I got on the 19.09 branch.
I compared the downloaded dependencies and it seems that on 19.09 quite a few more dependencies get downloaded, which are the same packages but with different versions, e.g.:
```diff
 go: finding github.com/xeipuuv/gojsonpointer v0.0.0-20180127040702-4e3ac2762d5f
 go: finding github.com/xeipuuv/gojsonreference v0.0.0-20180127040603-bd5ef7bd5415
 go: finding github.com/xeipuuv/gojsonschema v1.1.0
+go: finding golang.org/x/crypto v0.0.0-20180621125126-a49355c7e3f8
+go: finding golang.org/x/crypto v0.0.0-20180904163835-0709b304e793
+go: finding golang.org/x/crypto v0.0.0-20190211182817-74369b46fc67
+go: finding golang.org/x/crypto v0.0.0-20190308221718-c2843e01d9a2
+go: finding golang.org/x/crypto v0.0.0-20190418165655-df01cb2cc480
 go: finding golang.org/x/crypto v0.0.0-20190701094942-4def268fd1a4
 go: finding golang.org/x/exp v0.0.0-20190121172915-509febef88a4
+go: finding golang.org/x/lint v0.0.0-20181026193005-c67002cb31c3
+go: finding golang.org/x/lint v0.0.0-20190227174305-5b3e6a55c961
+go: finding golang.org/x/lint v0.0.0-20190301231843-5614ed5bae6f
+go: finding golang.org/x/lint v0.0.0-20190313153728-d0100b6bd8b3
 go: finding golang.org/x/lint v0.0.0-20190409202823-959b441ac422
```

I imagine this happens because 19.09 uses go 1.12 but master uses go 1.13, but I don't know for sure.
So I guess if this commit gets backported, the `modSha256` hash must be changed.

**Review by andrew-d — APPROVED** [review]

**Comment by bbigras** [comment]
Can this PR be updated to 3.2.0?

**Comment by timokau** [comment]
Superseded by #74371.

**Pull request closed without merging**